src/ipa_pc/mod.rs

```diff
@@ -940,13 +896,9 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
         labeled_polynomials: impl IntoIterator<Item = &'a LabeledPolynomial<G::ScalarField>>,
         commitments: impl IntoIterator<Item = &'a LabeledCommitment<Self::Commitment>>,
         query_set: &QuerySet<G::ScalarField>,
-        // TODO: In order to implement the Fiat-Shamir transform in a "clean" manner,
-        //       we need to pass a sponge instead of opening_challenges().
-        //       This sponge is then used to squeeze the opening challenge(s) as well
-        //       as the fresh query point x
-        opening_challenges: &dyn Fn(u64) -> G::ScalarField,
         rands: impl IntoIterator<Item = &'a LabeledRandomness<Self::Randomness>>,
         rng: Option<&mut dyn RngCore>,
+        fs_rng: &mut Self::RandomOracle,
```

Comment:
We should comment, that this implementation assumes that the commitments (as well as the query set) are already absorbed by the Fiat Shamir rng. This is a low-level function, one needs to be very careful in application.

Reply:
Done